domain/auth/usecase/authentication_test.go

```diff
@@ -0,0 +1,125 @@
+package usecase_test
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/francisleide/ChallengeGo/domain/auth"
+	"github.com/francisleide/ChallengeGo/domain/auth/usecase"
+	"github.com/francisleide/ChallengeGo/domain/entities"
+	"github.com/stretchr/testify/assert"
+	"golang.org/x/crypto/bcrypt"
+)
+
+func TestLogin(t *testing.T) {
+	t.Run("username and password are correct and no error is returned", func(t *testing.T) {
+		//prepare
+		mockRepo := new(auth.MockRepository)
+		authUC := usecase.NewAuthenticationUC(mockRepo)
+		accountOrigin := usecase.Credentials{
+			CPF:    "47708141001",
+			Secret: "abc123",
+		}
+		secret, _ := bcrypt.GenerateFromPassword([]byte(accountOrigin.Secret), bcrypt.DefaultCost)
+		accountReceived := entities.Account{
+			CPF:    accountOrigin.CPF,
+			Name:   "Pereira Silveira",
+			Secret: string(secret),
+		}
+
+		mockRepo.On("FindOne").Return(accountReceived, nil)
+
+		//test
+		err := authUC.Login(accountOrigin.CPF, accountOrigin.Secret)
+
+		//assert
+		assert.Equal(t, accountOrigin.CPF, accountReceived.CPF)
+		assert.NoError(t, err)
+		err = bcrypt.CompareHashAndPassword([]byte(accountReceived.Secret), []byte(accountOrigin.Secret))
```

Comment:
it's quite excessive this assertion since your code itself does it, won't you agree?